apps/root/src/pages/strategy-guardian-detail/strategy-management/deposit/tx-steps/index.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import TransactionSteps from '@common/components/transaction-steps';
+import { useEarnManagementState } from '@state/earn-management/hooks';
+
+const EarnTransactionSteps = ({ recapDataProps, ...rest }: Parameters<typeof TransactionSteps>[0]) => {
+  const { assetAmount } = useEarnManagementState();
+
+  const earnRecapDataProps = React.useMemo(
+    () => (recapDataProps ? { ...recapDataProps, assetAmount } : undefined),
+    [assetAmount]
```

Comment:
Same here, don't we why we subscribe here to redux and memoize it, if we can consume directly in `EarnDepositRecapData`

Reply:
I'm sorry i didnt understand the comment

Reply:
This is resolved now, it was in relation with accessing redux directly in the recap data component, but I agree its more useful now via props